client/src/services/ajax.js

```diff
@@ -0,0 +1,21 @@
+import {paramsObjectToQueryString} from "./formatting";
+
+export const sendHttpRequest = async (url, method, body, params, headers={}, callbackError) => {
+        try{
+            url += paramsObjectToQueryString(params);
+            return await fetch(url, {
+                method,
+                headers,
+                body
+            });
+        }
+        catch (e) {
+            callbackError(e);
+        }
+
+};
+
+export const handleResponse = async (response, statusOptions) => {
```

Comment:
I think `statusOptions` should be called `statusBasedActions` since it's an object that maps a `statusCode` to an action based on said `statusCode`

Reply:
what about `statusActions` ?